packages/zoe/tools/manualTimer.js

```diff
@@ -1,37 +1,79 @@
+// @ts-check
+
 import { E } from '@agoric/eventual-send';
+import { makeStore } from '@agoric/store';
+
+import './types';
+
+/* eslint-disable jsdoc/valid-types */
+/**
+ * @typedef {Object} ManualTimerAdmin
+ * @property {(msg?: string) => Promise<void>} tick
+ */
+/* eslint-enable jsdoc/valid-types */
 
-// A fake clock that also logs progress in tests.
+/**
+ * @typedef {ManualTimerAdmin & TimerService} ManualTimer
+ */
+
+/**
+ * A fake clock that also logs progress.
+ *
+ * @param {(...args: any[]) => void} log
+ * @param {Timestamp} [startValue=0]
+ * @returns {ManualTimer}
+ */
 export default function buildManualTimer(log, startValue = 0) {
   let ticks = startValue;
-  const schedule = new Map();
+
+  /** @type {Store<Timestamp, Array<TimerServiceHandler>>} */
+  const schedule = makeStore('Timestamp');
+
+  /** @type {ManualTimer} */
   const timer = {
-    setWakeup(deadline, handler) {
-      if (deadline <= ticks) {
-        log(`&& task was past its deadline when scheduled: ${deadline} &&`);
-        handler.wake(ticks);
-        return undefined;
-      }
-      log(`@@ schedule task for:${deadline}, currently: ${ticks} @@`);
-      if (!schedule.has(deadline)) {
-        schedule.set(deadline, []);
-      }
-      schedule.get(deadline).push(handler);
-      return deadline;
-    },
     // This function will only be called in testing code to advance the clock.
     async tick(msg) {
       ticks += 1;
       log(`@@ tick:${ticks}${msg ? `: ${msg}` : ''} @@`);
       if (schedule.has(ticks)) {
+        const handlers = schedule.get(ticks);
+        schedule.delete(ticks);
         await Promise.allSettled(
-          schedule.get(ticks).map(h => {
+          handlers.map(h => {
             log(`&& running a task scheduled for ${ticks}. &&`);
             return E(h).wake(ticks);
           }),
         );
       }
     },
-    createRepeater(delaySecs, interval) {
+    getCurrentTimestamp() {
+      return ticks;
+    },
+    setWakeup(baseTime, handler) {
+      if (baseTime <= ticks) {
+        log(`&& task was past its deadline when scheduled: ${baseTime} &&`);
+        handler.wake(ticks);
+        return undefined;
+      }
+      log(`@@ schedule task for:${baseTime}, currently: ${ticks} @@`);
+      if (!schedule.has(baseTime)) {
+        schedule.init(baseTime, []);
+      }
+      schedule.get(baseTime).push(handler);
+      return baseTime;
+    },
+    removeWakeup(handler) {
+      const baseTimes = [];
+      for (const [baseTime, handlers] of schedule.entries()) {
+        const index = handlers.indexOf(handler);
+        if (index >= 0) {
+          handlers.splice(index, 1);
+          baseTimes.push(baseTime);
+        }
+      }
+      return baseTimes;
```

Comment:
hmm, how is schedule being updated? handlers.splice edits in place through the use of side effects?
Why return baseTimes?

Reply:
> hmm, how is schedule being updated? handlers.splice edits in place through the use of side effects?

Correct.  I will also make it explicitly delete the schedule entry when `handlers` is empty.

> Why return baseTimes?

Ask @Chris-Hibbert.  That's his documented API that I'm implementing.

Reply:
> > hmm, how is schedule being updated? handlers.splice edits in place through the use of side effects?
> 
> Correct. I will also make it explicitly delete the schedule entry when `handlers` is empty.

I clarified it further.  No more side effects.